models/user.go

```diff
@@ -135,6 +136,7 @@ type User struct {
 	AllowImportLocal        bool // Allow migrate repository by local path
 	AllowCreateOrganization bool `xorm:"DEFAULT true"`
 	ProhibitLogin           bool `xorm:"NOT NULL DEFAULT false"`
+	LoginFailures           int  `xorm:"NOT NULL DEFAULT 0"`
```

Comment:
I worry about attaching this to a specific user as 1. this could allow an attacker to prevent a legitimate user from logging in, and 2. an attacker could "credential stuff" where they have a list of many users' passwords and try them all. This would still be many requests, but would be for multiple users.

This approach also doesn't time limit failures, so someone could write a script that calls the API and loops through all the users to lock out all users from the site.

An alternative approach could be based on IP, while it doesn't prevent someone from using a VPN to get different IPs, this still makes it harder for an attacker to bruteforce the site.

Reply:
With this PR, if someone write script/bot, he can only try 3 passwords. after must input captcha

Reply:
@techknowlogick this does not lockout user, just asks captcha, so legitimate users can still login.

Through per-IP counter in memory `map[string]struct{count, lastAuthTime}` could probably be better, that would prevent bruteforcing knownlist of user/psw